specification/src/main/asciidoc/chapters/Sparkplug_5_Operational_Behavior.adoc

```diff
@@ -383,16 +390,217 @@ published to the Edge Node using a NCMD Sparkplug message.
+
+* [tck-testable tck-id-operational-behavior-data-publish-dbirth]#[yellow-background]*DBIRTH messages
+MUST include all metrics for the specified Device that will ever be published for that Device within
+the established Sparkplug session.*#
+* [tck-testable tck-id-operational-behavior-data-publish-dbirth-values]#[yellow-background]*DBIRTH
+messages MUST include current values for all metrics.*#
+* [tck-testable tck-id-operational-behavior-data-publish-dbirth-change]#[yellow-background]*DDATA
+messages MUST only be published when Device level metrics change.*#
+** In other words, metric values that have not changed within the same Sparkplug Session MUST not be
+resent until a new Sparkplug session is established.
+* NDATA messages SHOULD be aggregated to include multiple metrics.
+** This is up to the application developer in terms of how many metrics should be aggregated in a
+single message, but it typically doesn't make sense to publish an MQTT message for every single
+metric change.
+** Multiple value changes for the same metric MAY be included in the same Sparkplug DDATA message as
+long as they have different timestamps.
+* [tck-testable tck-id-operational-behavior-data-publish-dbirth-order]#[yellow-background]*For all
+metrics where is_historical=false, DBIRTH and DDATA messages MUST keep metric values in
+chronological order in the list of metrics in the payload.*#
 
 [[operational_behavior_commands]]
 === Commands
-// TODO: Complete section
+
+Commands are used in Sparkplug to allow Sparkplug Host Applications to send data to Sparkplug Edge
+Nodes. Examples include writing to outputs of Sparkplug Edge Nodes and Devices or to request
+Rebirths from Edge Nodes. Custom command endpoints can be declared in an NBIRTH or DBIRTH message by
+an Edge Node or Device that may support functionality such as rebooting an Edge Node or Device. This
+is up to the Sparkplug implementor to define what functionality can be exposed.
+
+Security and access is an important aspect of commands. It may be the case that not all Sparkplug
+Host Applications should have the ability to send commands. This can be be controlled in multiple
+ways. ACLs (Access Control Lists) may be used to allow/disallow certain MQTT clients from publishing
+NCMD and DCMD messages. Security features in the Sparkplug Host Application itself could be used to
+allow/disallow certain users or applications from sending certain commands. Security features in the
+Sparkplug Edge Node application could be used to allow/disallow CMD messages to be honored. There
+are a number of ways in which this can be done and should be considered. However, implementation
+details are not covered in the Sparkplug Specification and is left to specific application designers
+to consider.
+
+There are two types of command (CMD) verbs in Sparkplug. There are NCMD and DCMD messages which
+denote the target command message is either for an Edge Node or Device respectively.
```

Comment:
This doesn't read quite right to me.  

"There are NCMD and DCMD messages which target Edge Nodes and Devices respectively"

Something like that.

Reply:
Good call - fixed.